server/src/test/java/org/opensearch/cluster/routing/allocation/BalanceConfigurationTests.java

```diff
@@ -380,6 +413,32 @@ public void testGlobalPrimaryBalance() throws Exception {
         verifyPrimaryBalance(clusterState);
     }
 
+    /**
+     * This test verifies global balance by creating indices iteratively and verify primary shards do not pile up on one
+     * @throws Exception generic exception
+     */
+    public void testGlobalPrimaryBalanceWithNodeDrops() throws Exception {
+        final float buffer = 0.05f;
+        AllocationService strategy = createAllocationService(getSettingsBuilderForPrimaryReBalance().build(), new TestGatewayAllocator());
+        ClusterState clusterState = ClusterState.builder(CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY)).build();
+        clusterState = addNode(clusterState, strategy);
+        clusterState = addNode(clusterState, strategy);
+        clusterState = addNode(clusterState, strategy);
+        clusterState = addNode(clusterState, strategy);
+        clusterState = addNode(clusterState, strategy);
```

Comment:
isn't there a method that allows adding multiple nodes without duplication the line multiple times? If not, lets create one.